spec/system/admin_branding_spec.rb

```diff
@@ -6,206 +6,266 @@
+          branding_page.logo_form.upload_image(image_type, image_file)
+        end
 
-      social_media_section_logos.each do |image_type|
-        expect(branding_page.logo_form.image_uploader(image_type)).to have_uploaded_image
-      end
+        social_media_section_logos.each do |image_type|
+          expect(branding_page.logo_form.image_uploader(image_type)).to have_uploaded_image
+        end
 
-      branding_page.logo_form.submit
-      expect(branding_page.logo_form).to have_saved_successfully
+        branding_page.logo_form.submit
+        expect(branding_page.logo_form).to have_saved_successfully
 
-      visit("/")
-      branding_page.visit
-      branding_page.logo_form.expand_social_media_section
+        visit("/")
+        branding_page.visit
+        branding_page.logo_form.expand_social_media_section
 
-      social_media_section_logos.each do |image_type|
-        expect(branding_page.logo_form.image_uploader(image_type)).to have_uploaded_image
+        social_media_section_logos.each do |image_type|
+          expect(branding_page.logo_form.image_uploader(image_type)).to have_uploaded_image
+        end
       end
-    end
 
-    it "can remove images" do
-      social_media_section_logos.each do |image_type|
-        SiteSetting.send("#{image_type}=", image_upload)
+      it "can remove images" do
+        social_media_section_logos.each do |image_type|
+          SiteSetting.send("#{image_type}=", image_upload)
+        end
+
+        branding_page.visit
+        branding_page.logo_form.expand_social_media_section
+
+        social_media_section_logos.each do |image_type|
+          expect(branding_page.logo_form.image_uploader(image_type)).to have_uploaded_image
+        end
+
+        social_media_section_logos.each do |image_type|
+          branding_page.logo_form.remove_image(image_type)
+        end
+
+        branding_page.logo_form.submit
+        expect(branding_page.logo_form).to have_saved_successfully
+
+        social_media_section_logos.each do |image_type|
+          expect(SiteSetting.send(image_type)).to eq(nil)
+        end
       end
+    end
+  end
 
+  describe "fonts" do
+    it "allows to change base and heading font" do
       branding_page.visit
-      branding_page.logo_form.expand_social_media_section
+      branding_page.fonts_form.select_font("base", "helvetica")
 
-      social_media_section_logos.each do |image_type|
-        expect(branding_page.logo_form.image_uploader(image_type)).to have_uploaded_image
-      end
+      expect(branding_page.fonts_form).to have_no_font("heading", "Oswald")
+      branding_page.fonts_form.show_more_fonts("heading")
+      branding_page.fonts_form.select_font("heading", "oswald")
 
-      social_media_section_logos.each do |image_type|
-        branding_page.logo_form.remove_image(image_type)
-      end
+      branding_page.fonts_form.submit
+      expect(branding_page.fonts_form).to have_saved_successfully
+
+      visit "/"
+      branding_page.visit
```

Comment:
```suggestion
      branding_page.visit
```

No need to visit root first.